src/Generics/SOP/BasicFunctors.hs

```diff
@@ -316,3 +332,108 @@ instance (Functor f, Show1 f, Show1 g) => Show1 (f :.: g) where
 -- | Extract the contents of a 'Comp' value.
 unComp :: (f :.: g) p -> f (g p)
 unComp (Comp x) = x
+
+-- * Mapping functions
+
+-- Implementation note:
+--
+-- All of these functions are just type specializations of
+-- 'coerce'. However, we currently still support GHC 7.6
+-- which does not support 'coerce', so we write them
+-- explicitly.
+
+-- | Lift the given function.
+--
+-- @since 0.2.5.0
+--
+mapII :: (a -> b) -> I a -> I b
+mapII f (I a) = I (f a)
+{-# INLINE mapII #-}
```

Comment:
the only thing I wonder, what's the `{-# INLINE -#}` behavior difference here.
Would `mapII = \f (I a) -> I (f a)` be closer to the behavior of `mapII = coerce`?